miniapps/elasticity/elasticity_driver.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+// This miniapp solves a quasistatic solid mechanics problem assuming an elastic
+// material and no body forces.
+//
+// The equation
+//                   ∇⋅σ(∇u) = 0
+//
+// with stress σ is solved for displacement u.
+//
+//             +----------+----------+
+//   fixed --->|                     |<--- constant displacement
+//             |                     |
+//             +----------+----------+
+//
+// This miniapp uses an elasticity operator that allows for a custom material.
+// By default the NeoHookeanMaterial is used. A linear elastic material is also
+// provided. Based on these examples, other materials could be implemented.
+//
+// The implementation of NeoHookeanMaterial also demenstrates the use of
+// automatic differentiation using either a native dual number implementation or
+// leveraging the Enzyme third party library.
+
+#include <mfem.hpp>
+
+#include "materials/linear_elastic.hpp"
+#include "materials/neohookean.hpp"
+#include "operators/elasticity_gradient_operator.hpp"
+#include "operators/elasticity_operator.hpp"
+#include "preconditioners/diagonal_preconditioner.hpp"
+
+using namespace std;
+using namespace mfem;
+
+/// This example only works in 3D.
+constexpr int dimension = 3;
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi;
+   int myid = mpi.WorldRank();
```

Comment:
```suggestion
   Mpi::Init(argc, argv);
```

and can just use `Mpi::Root()` instead of `myid == 0` below